doc/design/embargo-mvp.md

```diff
@@ -0,0 +1,131 @@
+Confirming will lock the dandiset for the duration of the release.
+Once the release finishes, the dandiset will be like any other unpublished draft dandiset.
+
+Once created, an embargoed dandiset is only visible or searchable to owners.
+Instead of the `draft` chip, an `embargoed` chip should be used instead.
+
+## CLI
+The CLI experience will be basically unchanged.
+Uploads to an embargoed dandiset will function exactly the same from an API perspective, the data simply goes to a different location.
+
+### TODO: how to browse/download embargoed zarr files?
+Credentials need to be generated, distributed, and used by the client.
+
+
+# Data storage
+Embargoed assets will be stored in a separate S3 bucket.
+This bucket is private and not browseable by the general public.
+
+The API server will use the embargo bucket in exactly the same way it uses the public bucket, but it will only store embargoed data.
+
+When releasing an embargoed dandiset, all asset data for that dandiset is copied to the public bucket.
+
+When uploading a new asset to an embargoed dandiset, the server will first check if that blob has already been uploaded publically.
+If so, the public blob will be used instead of uploading the data again to the embargo bucket.
+
+
+# Django changes
+
+## Models
+The `Dandiset` model will have an `embargo_status` field that is one of `PRIVATE`, `RELEASING`, or `PUBLIC`.
+* `PUBLIC` means that the Dandiset is publically accessible and publishable.
+  This is the state all Dandisets currently have.
+* `PRIVATE` means that the Dandiset is embargoed.
+  It is not searchable or viewable to non-owners.
+  It is not publishable.
+  Manifest YAML/JSON files will be written to the embargo bucket rather than the public bucket.
+* `RELEASING` means that the Dandiset is currently transitioning from embargoed to public.
+  All modification operations will return a 400 error while the release is in progress.
+  This includes metadata changes and uploads.
+
+A new `EmbargoedAssetBlob` model will be added.
+This behaves the same as a normal `AssetBlob`, but points to the embargo bucket rather than the public bucket.
+It also contains a reference to the `Dandiset` it belongs to.
+
+`Asset`s will point to exactly one `AssetBlob` or `EmbargoedAssetBlob`.
+An `Asset` will be considered "embargoed" if it has an `EmbargoedAssetBlob`.
+An embargoed `Asset` can only belong to the dandiset named in the `EmbargoedAssetBlob`.
+The automatically populated `Asset` metadata will also set the `access` field to the appropriate value depending on whether or not the asset is embargoed.
+
+The `Upload` model will have a new optional field `embargoed_dandiset` that points to an embargoed dandiset.
+If specified, the uploaded data will be sent to the embargoed bucket instead of the public bucket, and the `Upload` will create an `EmbargoedAssetBlob` when finalized.
```

Comment:
i've thought about one generic change related to this. that upload should require the dandiset id to be sent as a parameter when initiating. this would let allow us to use that information and would not allow an upload without creating a dandiset (since the token authorization must give the uploader ownership on the dandiset).

Reply:
this would require a breaking change to the CLI, but i think this is an easy change on both the api and cli side.

Reply:
didn't quite see this around ~115. my suggestion was simply that this would be a generic endpoint change to require dandiset as a parameter to the upload initialize endpoint.

Reply:
The relevant text can be seen [here](https://github.com/dandi/dandi-api/blob/ad8ac167cd27004de0119b54f9833e607cebf52c/doc/design/embargo-mvp.md?plain=1#L123-L133). This proposal keeps the parameter optional; if we want to make it required that would be out of scope for this MVP effort. I have filed #628 to track this.

Reply:
(Resolving this thread; please unresolve if you want to continue the discussion, or feel free to discuss further at #628.)